.github/actions/TestApplicationManifestConsistency/action.ps1

```diff
@@ -0,0 +1,100 @@
+function Test-ApplicationManifestConsistency
+{
+    param(
+        [Parameter(Mandatory = $false)]
+        [string]$ExpectedAppVersion = (GetApplicationVersion),
+        [Parameter(Mandatory = $false)]
+        [string]$ExpectedPlatformVersion = (GetPlatformVersion)
+    )
+    $errors = @()
+
+    $bcAppsManifests = (Get-ChildItem "src" -Recurse -Include app.json).FullName
+    foreach($path in $bcAppsManifests)
+    {
+        $manifest = Get-Content $path | ConvertFrom-Json
+        if ($ExpectedAppVersion -ne $manifest.version)
+        {
+            $errors += "ERROR: Wrong application version in manifest $path. Expected: $ExpectedAppVersion. Actual: $($manifest.version)"
+        }
+        foreach($item in $manifest.dependencies)
+        {
+            if ($ExpectedAppVersion -ne $item.version)
+            {
+                $errors += "ERROR: Wrong dependency version ($($item.name)) in manifest $path. Expected: $ExpectedAppVersion. Actual: $($item.version)"
+            }
+        }
+        if ($manifest.platform -and ($ExpectedPlatformVersion -ne $manifest.platform))
+        {
+            $errors += "ERROR: Wrong platform version in manifest $path. Expected: $ExpectedPlatformVersion. Actual: $($manifest.platform)"
+        }
+        if ($manifest.publisher -ne "Microsoft")
+        {
+            # Allow Partner publisher if path contains 'Partner Test' and is under App\BCApps
+            if ($path -like "*Partner Test*" -and $manifest.publisher -eq "Partner")
+            {
+                Write-Host "Allowing Partner publisher in $path as it contains 'Partner Test' and is under BCApps."
+            }
+            else
+            {
+                $errors += "ERROR: Wrong publisher in manifest $path. Expected: Microsoft. Actual: $($manifest.publisher)"
```

Comment:
If the path contains "Partner Test", but the publisher isn't "Partner", this message will be wrong.

The check needs more attention.

Reply:
Idk... I'd say the expected publisher should always be Microsoft regardless of the path. In this one app we allowed it to be something else but that is an exception.

Reply:
That's fine as well.

My understanding was the _publisher_ should be "Partner" for apps in the "Partner Test" folder.